### PR TITLE
Replace select controls with new SelectControl (Autocomplete) component

### PR DESCRIPTION
**Pull request by joshuatf**
Fixes (partially) #2834

* Adds the `isSearchable` prop to the autocomplete component and renames it to `SelectControl` to cover more use cases.
* Allows strings for simple selects `selected` values in addition to arrays.
* Replaces instances of `SimpleSelectControl` with this component.
* Replaces the country dropdown with this component.

### Screenshots
Screen Shot 2019-10-03 at 1 58 29 AM
Screen Shot 2019-10-03 at 1 51 14 AM

### Detailed test instructions:

1. Enable the onboarding profiler and visit both store details and business details.
2. Make sure the country dropdown and all dropdowns on the business details step continue to work as expected.
3.  Make sure that submitted/changed values for dropdowns continue working as expected.

### Changelog Note:

Dev: Replace applicable select components with searchable SelectControl component.

### CODE REVIEW

File: client/dashboard/profile-wizard/style.scss [hunk 1]
```diff
@@ -251,6 +251,37 @@
 		}
 	}
 
+	.woocommerce-select-control__control {
+		margin: $gap 0;
+
+		&.is-active {
+			box-shadow: 0 0 0 1px #673d99;
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/dashboard/profile-wizard/style.scss]: Updated in https://github.com/woocommerce/woocommerce-admin/pull/2997/commits/aa3fb5a700dd76d7f8a02343857828507a32128e

File: client/dashboard/profile-wizard/style.scss [hunk 2]
```diff
@@ -251,6 +251,37 @@
...
+			font-family: dashicons, sans-serif;
+			font-size: 20px;
+			content: '\f140';
+			z-index: 1;
+			height: 24px;
+			width: 24px;
+			margin-top: 0;
+			top: 0;
+			right: $gap;
+			bottom: 16px;
```
[the remark this answers is not in the document]
Comment (joshuatf) [reply client/dashboard/profile-wizard/style.scss]: Updated in https://github.com/woocommerce/woocommerce-admin/pull/2997/commits/aa3fb5a700dd76d7f8a02343857828507a32128e

**Comment by joshuatf** [comment]
Thanks for the review, @justinshreve!

>After selecting an option, it feels like I should be able to click the down arrow to pull up the dropdown mode again. The only way to get it back is to erase the contents.

This is an interesting point.  It seems like this is somewhat expected behavior based on the combobox example [here](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html).

However, since we have the prop `excludeSelectedOptions` nothing should show when refocusing the input box unless this prop is `false`.  The up/down should now work if an option is selected when a query has multiple options before blur and refocus of the input box.

Open to input on this behavior, but made some worthwhile refactor changes in https://github.com/woocommerce/woocommerce-admin/pull/2997/commits/51abef000d8bd86fb4b4a1ba1d6dba672a2abd38 to make this a bit easier to handle.

>I also think we want to bump the packages changelog (and version too, since two components get removed).

Added changelog notes and bumped version in `packages.json`- anything else we need to do here?

**Comment by justinshreve** [comment]
> This is an interesting point. It seems like this is somewhat expected behavior based on the combobox example here.

Thanks for that link. It does look like the component is following the listbox combo behaviors. I'd say let's stick with this behavior as you have it implemented. I was initially expecting all the countries to show back up if I clicked the arrow, but I agree if things got filtered down that doesn't make much sense.

Changes are looking good, but it looks like some tests need adjusted.

File: packages/components/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,8 @@
+# 4.1.0
```
Comment (justinshreve) [inline packages/components/CHANGELOG.md]: I believe we've marked this as `# 4.1.0 (Unreleased)` in the passed, so we know quickly where to add additional items.

I think this is all we need to do for now, until we decide to do a components release.

Comment (joshuatf) [reply packages/components/CHANGELOG.md]: Updated!

**Comment by joshuatf** [comment]
@justinshreve Fixed those failing tests in https://github.com/woocommerce/woocommerce-admin/pull/2997/commits/39142c083622fa21620796c0b2175404f714b422 and added a new one for the simple select.

**Pull request merged**